### DISCUSSION TITLE
Why does GitX take control of `.git` extensions and can we prevent it?

### DISCUSSION THREAD

**Discussion by hugocf** [gitx/gitx#408] [Q&A]
Is there any strong reason for GitX to register the extension `.git` for itself?
https://github.com/gitx/gitx/blob/master/Resources/Info.plist#L20-L24

Many times I name my local clones `example.git` and oppose to others like `example.hs` to indicate which vcs it uses.

Trying to `open .` a folder from the Termina when inside the local repo results in opening GitX instead of Finder, which is what I would expect in every other circumstance.

Is there something we could do about this or is the decision to register the extension `.git`  a deliberate and strongly held one?  
_(I tried looking for discussions or issues about it but found none)_

**Comment by ADTC** [comment]
I think GitX is trying to make it easy to open a Git repo from the hidden `.git` folder inside it, but this conflicts with your naming scheme.

Personally I put all my Git repos inside a folder called `Git`, so it would be `~/Git/example` instead of `~/example.git`.

**Comment by MJCollett** [comment]
I think it's a fairly general rule that a 'normal' folder should have no extension.  If you give it an extension you are inviting the OS to treat it as a package instead.  So I'm not inclined to attach specific blame to GitX in this case.  I only use the '.git' extension (not to be confused with the '.git' hidden filename!) for headless repositories.

**Reply by ADTC** [reply]
Yes, this is true. macOS can potentially treat any folder ending with `.app` as an Application package.
